usaspending_api/disaster/v2/views/object_class/spending.py

```diff
@@ -0,0 +1,139 @@
+        "page_metadata": get_pagination_metadata(len(object_classes), pagination.limit, pagination.page),
+    }
+
+
+class ObjectClassSpendingViewSet(PaginationMixin, SpendingMixin, DisasterBase):
+    """View to implement the API"""
+
+    endpoint_doc = "usaspending_api/api_contracts/contracts/v2/disaster/object_class/spending.md"
+
+    @cache_response()
+    def post(self, request):
+        if self.spending_type == "award":
+            results = list(self.award_queryset)
+        else:
+            results = list(self.total_queryset)
+
+        return Response(construct_response(results, self.pagination))
+
+    @property
+    def total_queryset(self):
+        filters = [
+            Q(
+                Q(obligations_incurred_by_program_object_class_cpe__gt=0)
+                | Q(obligations_incurred_by_program_object_class_cpe__lt=0)
+                | Q(gross_outlay_amount_by_program_object_class_cpe__gt=0)
+                | Q(gross_outlay_amount_by_program_object_class_cpe__lt=0)
+            ),
+            Q(disaster_emergency_fund__in=self.def_codes),
+            Q(object_class__isnull=False),
+            self.all_closed_defc_submissions,
+        ]
+
+        annotations = {
+            **self.universal_annotations(),
+            "obligation": Coalesce(
+                Sum(
+                    Case(
+                        When(
+                            self.final_period_submission_query_filters,
+                            then=F("obligations_incurred_by_program_object_class_cpe"),
+                        ),
+                        default=Value(0),
+                    )
+                ),
+                0,
+            ),
+            "outlay": Coalesce(
+                Sum(
+                    Case(
+                        When(
+                            self.final_period_submission_query_filters,
+                            then=F("gross_outlay_amount_by_program_object_class_cpe"),
+                        ),
+                        default=Value(0),
+                    )
+                ),
+                0,
+            ),
+        }
+
+        # Assuming it is more performant to fetch all rows once rather than
+        #  run a count query and fetch only a page's worth of results
+        return (
+            FinancialAccountsByProgramActivityObjectClass.objects.filter(*filters)
+            .values("object_class__major_object_class", "object_class__major_object_class_name",)
+            .annotate(**annotations)
+            .values(*annotations.keys())
+        )
+
+    @property
+    def award_queryset(self):
+        filters = [
+            Q(disaster_emergency_fund__in=self.def_codes),
+            Q(object_class__isnull=False),
+            self.all_closed_defc_submissions,
+        ]
+
+        annotations = {
+            **self.universal_annotations(),
+            "obligation": Coalesce(Sum("transaction_obligated_amount"), 0),
+            "outlay": Coalesce(
```

Comment:
pre-existing property for outlay annotation: `self.outlay_field_annotation` in FabaOutlayMixin